src/adapters/book-adapters.ts

```diff
@@ -0,0 +1,254 @@
+import { setCookie, getCookie } from "../util/Authentication"
+export const fetchAPIData = async (url: string, method: string, body:object | undefined , headers?:object) => {
+    const cookie = getCookie('shelf-share-session')as { token: string }
+    const token = cookie?.token
+    const response = await fetch(url , {
+        method,
+        body: JSON.stringify(body),
+        headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-Type': 'application/json',
+            ...headers,
+        },
+    })
+    
+    if(response.status.toString().startsWith('4')||response.status.toString().startsWith('5')){
+      console.log("Error", response.status)
+    }
+    else{
+      const data = await response.json();  
+      return data; 
+    }
+        
+}
```

Comment:
Import this from fetch.ts. Like you said, you will need to change fetch.ts so that body can be undefined.

Reply:
@jpremo  Thank you for the feedback. Actually, I have imported fetch.ts and added body undefined to fetch.ts then when we checked the login route it gave us an error "JSON invalid/undefined....." that's why I kept this fetch file on my book-adapters, could you please advise on this issue?

Reply:
The function is an exact copy from fetch.ts. The login route wouldn't use the fetchAPIData defined here since it is in the auth-adapters.ts file. I tested this locally with the import and it seemed fine to me. Let me know if you can still replicate the issue and if so provide a screenshot of the entire error.